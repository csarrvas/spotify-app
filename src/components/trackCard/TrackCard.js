import React from 'react';
import { millisToMinutesAndSeconds } from '../../helpers/helpers';
import './trackCard.scss';

const TrackCard = ({ track, fromAlbum = true, topNumber = 0 }) => {
  return (
    <div className="track">
      {fromAlbum
        ? <p>{`CD-${track.disc_number} / ${track.track_number} - ${track.name}`}</p>
        : <p>{`${topNumber} - ${track.name}`}</p>
      }
      <p>{`Duration: ${millisToMinutesAndSeconds(track.duration_ms)}`}</p>
      <a target="_blank" rel="noopener noreferrer" href={track.external_urls.spotify}>
        <i className="fas fa-play-circle">Go to this song on spotify</i>
      </a>
    </div>
  );
}
 
export default TrackCard;