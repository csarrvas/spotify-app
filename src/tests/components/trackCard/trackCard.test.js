import React from 'react';
import { shallow } from 'enzyme';
import TrackCard from '../../../components/trackCard/TrackCard';
import { track } from '../../fixtures/data';

test('should render TrackCard correctly', () => {
  const wrapper = shallow(<TrackCard track={track}/>);
  expect(wrapper).toMatchSnapshot();
});