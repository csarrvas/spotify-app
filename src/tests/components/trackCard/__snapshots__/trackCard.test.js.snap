// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TrackCard correctly 1`] = `
<div
  className="track"
>
  <p>
    CD-1 / 2 - Give Me Everything (feat. Ne-Yo, Afrojack & Nayer)
  </p>
  <p>
    Duration: 4:12
  </p>
  <a
    href="https://open.spotify.com/track/4QNpBfC0zvjKqPJcyqBy9W"
    rel="noopener noreferrer"
    target="_blank"
  >
    <i
      className="fas fa-play-circle"
    >
       Go to this song on spotify
    </i>
  </a>
</div>
`;
